lib/matplotlib/axes/_axes.py

```diff
@@ -3846,10 +3846,9 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
         if extent is not None:
             xmin, xmax, ymin, ymax = extent
         else:
-            xmin = np.amin(x)
-            xmax = np.amax(x)
-            ymin = np.amin(y)
-            ymax = np.amax(y)
+            xmin, xmax = (np.amin(x), np.amax(x)) if x.any() else (0, 1)
```

Comment:
won't `x.any()` return `False` on `np.zeros(5)`?  I think this should be `.... if len(x) else ...`